### DISCUSSION TITLE
Getting permission denied error when updating birdnet-pi in local network GUI

### DISCUSSION THREAD

**Discussion by pmoenart** [mcguirepr89/BirdNET-Pi#837] [General]
Hey y'all. I've been getting an error when hitting update on my birdnet-pi. I wanted to know if anyone else has encountered this before. When I click update, this is returned:

update_birdnet.sh: Permission denied

Thank you

**Comment by gerdesj** [comment]
Where on earth to start?  Did you install your BNP by installing RaspberryPi OS lite and then run the one liner install and that's it?  How old is it?

**Reply by pmoenart** [reply]
Yeah that's all I did. It's a couple weeks old.

**Reply by gerdesj** [reply]
Mine too and I've just updated it via Tools -> System Controls.

Can you login via Tools -> Web Terminal and run _sudo -i_ ? If so:

_apt update
apt upgrade
apt autoremove
reboot_
 
That will update the OS but not BNP.  Try that again after the reboot.

**Reply by pmoenart** [reply]
I did all that and I still get the same error. The OS did update. Am I able to start a BNP update from the web terminal or would that not make a difference?

**Reply by gerdesj** [reply]
I'm fairly (and only fairly!) sure that if you run the original install command gain it will update itself.  However I am still getting to grips with the code so I suggest that given you have a reasonably up to date BNP and your OS is fully patched you are in a good place.  

You can always update your OS like that and unlike some other OSs  that means that everything is updated except for off piste stuff.  BNP is a bit off piste but the components it requires will still be updated via apt.

**Reply by pmoenart** [reply]
I have about 2000 commits that I am behind. I feel like that is a lot but I am very new to all of this.

**Reply by ehpersonal38** [reply]
Hi, please check the issues - https://github.com/mcguirepr89/BirdNET-Pi/issues/833

This command will fix the perms for you:
`chmod a+x ~/BirdNET-Pi/scripts/update_birdnet.sh `

**Reply by southsko** [reply]
> Hi, please check the issues - #833
> 
> This command will fix the perms for you: `chmod a+x ~/BirdNET-Pi/scripts/update_birdnet.sh `

That fixed it for me!

**Reply by jflyons** [reply]
Very helpful for me too!
